app/scripts/modules/core/src/presentation/icons/Icon.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { memo } from 'react';
+
+import { iconsByName } from './iconsByName';
+
+export type IconNames = keyof typeof iconsByName;
+
+export type IIconProps = {
+  name: IconNames;
+  appearance?: 'light' | 'neutral' | 'dark';
+  size?: string;
```

Comment:
```suggestion
  size?: 'extraSmall' | 'small' | 'medium' | 'large' | 'extraLarge' | string;
```

Reply:
ha. You know what's funny is I wrote exactly this union out in my editor, and when I looked over the diff before committing I decided to take it out. At the time it felt like even though I enjoyed having it there we weren't getting much from it given TS coalesces the whole thing down to `string` (e.g. no autocomplete). You independently wishing it was there gives me the motivation to go with it 😄